src/nnet3/nnet-utils.cc

```diff
@@ -264,6 +280,57 @@ void SetLearningRate(BaseFloat learning_rate,
   }
 }
 
+void SetLearningRates(const Vector<BaseFloat> &learning_rates,
+                     Nnet *nnet) {
+  for (int32 c = 0, i = 0; c < nnet->NumComponents(); c++) {
+    Component *comp = nnet->GetComponent(c);
+    if (comp->Properties() & kUpdatableComponent) {
+      // For now all updatable components inherit from class UpdatableComponent.
+      // If that changes in future, we will change this code.
+      UpdatableComponent *uc = dynamic_cast<UpdatableComponent*>(comp);
+      if (uc == NULL)
+        KALDI_ERR << "Updatable component does not inherit from class "
+            "UpdatableComponent; change this code.";
+      KALDI_ASSERT(i < learning_rates.Dim());
+      uc->SetLearningRate(learning_rates(i));
```

Comment:
this code would not work correctly in neural nets that make use of the learning-rate-factor mechanism.
I think I need to make some changes to clarify things and make the interface more powerful before you can fix this- I'll try to do this today.